staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -269,37 +314,59 @@ func (cfgCtlr *configController) processNextWorkItem() bool {
 
 	func(obj interface{}) {
 		defer cfgCtlr.configQueue.Done(obj)
-		if !cfgCtlr.syncOne() {
+		report := cfgCtlr.SyncOne()
+		switch {
+		case report.NeedRetry:
 			cfgCtlr.configQueue.AddRateLimited(obj)
-		} else {
+		case report.Wait > 0:
+			cfgCtlr.configQueue.AddAfter(obj, report.Wait)
```

Comment:
Two questions:
1) an event will add a work item to the queue and that will force the sync again irrespective of whether or not the wait is done, right?
2) for our purposes, is there a difference between `AddRateLimited(obj)/AddAfter(obj)` and `AddRateLimited(0)/AddAfter(0)`

Reply:
1. If by "event" you mean a notification from an informer (i.e., a call to one of its `cache.ResourceEventHandlerFuncs`) then the answer is yes.  And this is true for both `AddRateLimited` and `AddAfter` (in fact, the implementation of `AddRateLimited` just calculates a delay and calls `AddAfter`).

2. No real difference.  `obj` will always be `0` in the current code, and breaking that would involve a lot of code changes.